Neos.Flow/Tests/Unit/Mvc/Routing/AttributeRoutesProviderTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Neos\Flow\Tests\Unit\Mvc\Routing;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Mvc\Routing;
+use Neos\Flow\Mvc\Routing\Route;
+use Neos\Flow\Mvc\Routing\Routes;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Flow\Reflection\ReflectionService;
+use Neos\Flow\Tests\UnitTestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Testcase for the MVC Web Routing Routes Class
+ */
+class AttributeRoutesProviderTest extends UnitTestCase
+{
+    private ReflectionService|MockObject $mockReflectionService;
+    private ObjectManagerInterface|MockObject $mockObjectManager;
+    private Routing\AttributeRoutesProvider $annotationRoutesProvider;
+
+    public function setUp(): void
+    {
+        $this->mockReflectionService = $this->createMock(ReflectionService::class);
+        $this->mockObjectManager = $this->createMock(ObjectManagerInterface::class);
+
+        $this->annotationRoutesProvider = new Routing\AttributeRoutesProvider(
+            $this->mockReflectionService,
+            $this->mockObjectManager,
+            ['Vendor\Example\Controller\ExampleController']
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function noAnnotationsYieldEmptyRoutes(): void
+    {
+        $this->mockReflectionService->expects($this->once())
+            ->method('getClassesContainingMethodsAnnotatedWith')
+            ->with(\Neos\Flow\Annotations\Route::class)
+            ->willReturn([]);
+
+        $routes = $this->annotationRoutesProvider->getRoutes();
+        $this->assertEquals(Routes::empty(), $routes);
+    }
+
+    /**
+     * @test
+     */
+    public function routesFromAnnotationAreCreatedWhenClassNamesMatch(): void
+    {
+        $this->mockReflectionService->expects($this->once())
+            ->method('getClassesContainingMethodsAnnotatedWith')
+            ->with(Flow\Route::class)
+            ->willReturn(['Vendor\Example\Controller\ExampleController']);
+
+        $this->mockReflectionService->expects($this->once())
+            ->method('getMethodsAnnotatedWith')
+            ->with('Vendor\Example\Controller\ExampleController', Flow\Route::class)
+            ->willReturn(['specialAction']);
+
+        $this->mockReflectionService->expects($this->once())
+            ->method('getMethodAnnotations')
+            ->with('Vendor\Example\Controller\ExampleController', 'specialAction', Flow\Route::class)
+            ->willReturn([
+                new Flow\Route(uriPattern: 'my/path'),
+                new Flow\Route(uriPattern: 'my/other/path', name: 'specialRoute', defaults: ['test' => 'foo'], httpMethods: ['get', 'post'])
```

Comment:
lowercase httpMethods will not work as we do case sensitive comparison here: https://github.com/neos/flow-development-collection/blob/c6e32474c0fa0ce1940c2d2e6fb9aa7f87f3d566/Neos.Flow/Classes/Mvc/Routing/Route.php#L490

we might as well validate that in the constructor